dlt/load/load.py

```diff
@@ -246,8 +247,14 @@ def get_completed_table_chain(
                 for job in table_jobs
             ):
                 return None
-            # if there are no jobs for the table, skip it, unless the write disposition is replace, as we need to create and clear the child tables
-            if not table_jobs and top_merged_table["write_disposition"] != "replace":
+            # if there are no jobs for the table, skip it, unless child tables need to be replaced
+            needs_replacement = False
+            if top_merged_table["write_disposition"] == "replace" or (
```

Comment:
why it is changed?

Reply:
This is needed to propagate deletes to child tables. If we provide only a primary key and the `hard_delete` column for a nested table, such as happens on lines 584 and 599 of `test_merge_disposition.py`, the child tables wouldn't get included in the table chain, and those deletes would only be executed on the root table.

Reply:
I still do not get it. We have jobs for this table because in both those lines we declare some data. The exception for replace is only for the case that there is no data at all. does not happen here. IMO you should try to remove it and find the problem elsewhere or ping me on slack to discuss it